pkg/agent/supportbundlecollection/support_bundle_controller_test.go

```diff
@@ -0,0 +1,266 @@
+	"antrea.io/antrea/pkg/support"
+)
+
+type fakeController struct {
+	*SupportBundleController
+	mockController *gomock.Controller
+}
+
+type antreaClientGetter struct {
+	clientset versioned.Interface
+}
+
+func (g *antreaClientGetter) GetAntreaClient() (versioned.Interface, error) {
+	return g.clientset, nil
+}
+
+func newFakeController(t *testing.T) (*fakeController, *fakeversioned.Clientset) {
+	controller := gomock.NewController(t)
+	clientset := &fakeversioned.Clientset{}
+	supportBundleController := NewSupportBundleController("vm1", controlplane.SupportBundleCollectionNodeTypeExternalNode, "vm-ns", &antreaClientGetter{clientset}, nil,
+		nil, nil, true, true)
+	supportBundleController.sftpUploader = &testUploader{}
+	return &fakeController{
+		SupportBundleController: supportBundleController,
+		mockController:          controller,
+	}, clientset
+}
+
+func TestSupportBundleCollectionAdd(t *testing.T) {
+	testcases := []struct {
+		name                    string
+		supportBundleCollection *cpv1b2.SupportBundleCollection
+	}{
+		{
+			name:                    "Add SupportBundleCollection",
+			supportBundleCollection: generateSupportbundleCollection("supportBundle1", "sftp://10.220.175.92:22/root/supportbundle"),
+		},
+		{
+			name:                    "Add SupportBundleCollection without url prefix",
+			supportBundleCollection: generateSupportbundleCollection("supportBundle2", "10.220.175.92:22/root/supportbundle"),
+		},
+	}
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			newAgentDumper = func(fs afero.Fs, executor exec.Interface, ovsCtlClient ovsctl.OVSCtlClient, aq agentquerier.AgentQuerier, npq querier.AgentNetworkPolicyInfoQuerier, since string, v4Enabled, v6Enabled bool) support.AgentDumper {
+				return &mockAgentDumper{}
+			}
+			defer func() {
+				newAgentDumper = support.NewAgentDumper
+			}()
+			controller, clientset := newFakeController(t)
+			var bundleStatus *cpv1b2.SupportBundleCollectionStatus
+			clientset.AddReactor("update", "supportbundlecollections/status", k8stesting.ReactionFunc(func(action k8stesting.Action) (bool, runtime.Object, error) {
+				bundleStatus = action.(k8stesting.UpdateAction).GetObject().(*cpv1b2.SupportBundleCollectionStatus)
+
+				return false, bundleStatus, nil
+			}))
+			controller.addSupportBundleCollection(tt.supportBundleCollection)
+			assert.NoError(t, controller.syncSupportBundleCollection(tt.supportBundleCollection.Name))
+			assert.Equal(t, true, bundleStatus.Nodes[0].Completed)
+		})
+	}
+}
+
+func TestSupportBundleCollectionDelete(t *testing.T) {
+	controller, _ := newFakeController(t)
+	deletedBundle := generateSupportbundleCollection("deletedBundle", "sftp://10.220.175.92/root/supportbundle")
+	controller.addSupportBundleCollection(deletedBundle)
+	controller.deleteSupportBundleCollection(deletedBundle)
+	assert.NoError(t, controller.syncSupportBundleCollection("deletedBundle"))
+}
+
+func TestSupportBundleCollectionRetry(t *testing.T) {
```

Comment:
I feel this could be merged to TestSupportBundleCollectionAdd, you just need to parameterize sftpUploader

Reply:
Thanks, have updated.